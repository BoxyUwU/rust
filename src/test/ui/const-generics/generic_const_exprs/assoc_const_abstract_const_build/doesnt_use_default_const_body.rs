#![feature(generic_const_exprs)]
#![allow(incomplete_features)]

trait Trait<const N: usize> {
    const ASSOC: usize = N + 1;
}

fn foo<const N: usize, T: Trait<N>>() -> [(); <T as Trait<N>>::ASSOC] {
    [(); N + 1]
    //~^ error: mismatched types
    //~^^ error: unconstrained generic constant
}

fn main() {}
