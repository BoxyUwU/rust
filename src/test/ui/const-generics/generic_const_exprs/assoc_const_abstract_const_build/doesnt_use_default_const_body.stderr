error[E0308]: mismatched types
  --> $DIR/doesnt_use_default_const_body.rs:9:5
   |
LL |     [(); N + 1]
   |     ^^^^^^^^^^^ expected `<T as Trait<N>>::ASSOC`, found `N + 1`
   |
   = note: expected type `<T as Trait<N>>::ASSOC`
              found type `N + 1`

error: unconstrained generic constant
  --> $DIR/doesnt_use_default_const_body.rs:9:10
   |
LL |     [(); N + 1]
   |          ^^^^^
   |
   = help: try adding a `where` bound using this expression: `where [(); N + 1]:`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
